staging/src/k8s.io/dynamic-resource-allocation/structured/allocator.go

```diff
@@ -122,7 +127,7 @@ func (a *Allocator) Allocate(ctx context.Context, node *v1.Node) (finalResult []
 	defer alloc.logger.V(5).Info("Done with allocation", "success", len(finalResult) == len(alloc.claimsToAllocate), "err", finalErr)
 
 	// First determine all eligible pools.
-	pools, err := GatherPools(ctx, alloc.slices, node)
+	pools, err := GatherPools(ctx, alloc.slices, node, a.features.PartitionableDevices)
```

Comment:
Maybe just pass `a.features` so the signature doesn't have to change later if other features affect this function?